cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeUtils.java

```diff
@@ -262,4 +346,54 @@ protected void doStop() {
       }
     };
   }
+
+  /**
+   * Uploads the spark event logs through the runtime service.
+   */
+  private static void uploadEventLogs(File eventLogDir, SparkRuntimeContext runtimeContext) throws IOException {
+
+    ProgramRunId programRunId = runtimeContext.getProgramRunId();
+
+    // Find the event file to upload. There should only be one for the current application.
+    File eventFile = Optional.ofNullable(eventLogDir.listFiles())
+      .map(Arrays::stream)
+      .flatMap(Stream::findFirst)
+      .orElse(null);
+    LOG.debug("Uploading event file {} for program run {}", eventFile, programRunId);
+
+    RemoteClient remoteClient = new RemoteClient(runtimeContext.getDiscoveryServiceClient(), Constants.Service.RUNTIME,
+                                                 new DefaultHttpRequestConfig(false),
+                                                 Constants.Gateway.INTERNAL_API_VERSION_3 + "/runtime/namespaces/");
+    String path = String.format("%s/apps/%s/versions/%s/%s/%s/runs/%s/spark-event-logs/%s",
+                                programRunId.getNamespace(),
+                                programRunId.getApplication(),
+                                programRunId.getVersion(),
+                                programRunId.getType().getCategoryName(),
+                                programRunId.getProgram(),
+                                programRunId.getRun(),
+                                eventFile.getName());
+
+    Retries.runWithRetries(() -> {
+      // Stream out the messages
+      HttpURLConnection urlConn = remoteClient.openConnection(HttpMethod.POST, path);
+      try {
+        urlConn.setChunkedStreamingMode(CHUNK_SIZE);
+        urlConn.setRequestProperty(HttpHeaders.CONTENT_TYPE, "application/octet-stream");
```

Comment:
Maybe we can use javax.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM?

Reply:
fixed